ingestr/main_test.py

```diff
@@ -1538,3 +1545,188 @@ def test_db_to_db_exclude_columns(source, dest):
+            traceback.print_exception(*result.exc_info)
+            raise AssertionError(result.exception)
+
+
+    def smoke_test(dest, dynamodb):
+        dest_uri = dest.start()
+        dest_table = f"public.dynamodb_{get_random_string(5)}"
+        dest_engine = sqlalchemy.create_engine(dest_uri)
+
+        result = invoke_ingest_command(
+            dynamodb.uri,
+            dynamodb.db_name,
+            dest_uri,
+            dest_table,
+            "append",
+            "updated_at"
+        )
+
+        assert_success(result)
+        result = dest_engine.execute(f"SELECT id, updated_at from {dest_table} ORDER BY id").fetchall()
+        assert len(result) == 3
+        for i in range(len(result)):
+            assert result[i][0] == dynamodb.data[i]["id"]
+            assert result[i][1] == pendulum.parse(dynamodb.data[i]["updated_at"])
+            
+        dest.stop()
+
+    def append_test(dest, dynamodb):
+        dest_uri = dest.start()
+        dest_table = f"public.dynamodb_{get_random_string(5)}"
+
+        # connection pooling causes issues with duckdb, when the connection
+        # is reused below, so we disable pooling.
+        dest_engine = sqlalchemy.create_engine(dest_uri, poolclass=NullPool)
+
+        # we run it twice to assert that the data in destination doesn't change
+        for i in range(2):
+            result = invoke_ingest_command(
+                dynamodb.uri,
+                dynamodb.db_name,
+                dest_uri,
+                dest_table,
+                "append",
+                "updated_at"
+            )
+
+            assert_success(result)
+            result = dest_engine.execute(f"SELECT id, updated_at from {dest_table} ORDER BY id").fetchall()
+            assert len(result) == 3
+            for i in range(len(result)):
+                assert result[i][0] == dynamodb.data[i]["id"]
+                assert result[i][1] == pendulum.parse(dynamodb.data[i]["updated_at"])
+            
+        dest.stop()
+
+    def incremental_test(dest, dynamodb):
+        dest_uri = dest.start()
+        dest_table = f"public.dynamodb_{get_random_string(5)}"
+        dest_engine = sqlalchemy.create_engine(dest_uri, poolclass=NullPool)
+
+        result = invoke_ingest_command(
+            dynamodb.uri,
+            dynamodb.db_name,
+            dest_uri,
+            dest_table,
+            inc_strategy="merge",
+            inc_key="updated_at",
+            interval_start="2024-01-01T00:00:00",
+            interval_end="2024-02-01T00:01:00" # upto the second entry
+        )
+        assert_success(result)
+        rows = dest_engine.execute(f"SELECT id, updated_at from {dest_table} ORDER BY id").fetchall()
+        assert len(rows) == 2
+        for i in range(len(rows)):
+            assert rows[i][0] == dynamodb.data[i]["id"]
+            assert rows[i][1] == pendulum.parse(dynamodb.data[i]["updated_at"])
+
+        # ingest the rest
+        result = invoke_ingest_command(
```

Comment:
why not run another attempt once more, similar to the above, to check the data? also, I'd suggest having some tests for other strategies like append or delete insert.